### PR TITLE
try to fix DefaultResourceFinder

### PR DESCRIPTION
**Pull request by guwirth**

### CODE REVIEW

**Comment by guwirth** [comment]
Working but looks complicated. Don't know if there is a easier solution?

File: sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/xunit/DefaultResourceFinder.java [hunk 1]
```diff
@@ -22,17 +22,32 @@
...
+      files = fs.files(p.and(
+        p.hasLanguage("c++"), p.hasType(InputFile.Type.TEST), p.hasAbsolutePath(file.getPath())
+      ));
+    } else {
+      files = fs.files(p.and(
+        p.hasLanguage("c++"), p.hasType(InputFile.Type.TEST), p.hasRelativePath(file.getPath())
+      ));
+    }
+    org.sonar.api.resources.File unitTestFile = null;
+    if (files.iterator().hasNext()) {
```
Comment (guwirth) [inline sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/xunit/DefaultResourceFinder.java]: Is this ok to use the first hit?

**Comment by guwirth** [comment]
@wenns This solution works.



I'm not familiar with the new API so don't know if there is a easier solution. Maybe someone more familiar with predicates, FileSystem, can review it.

My understanding: FileSystem is seraching for the file, getResource in the database.
Also don't know if fromIOFile is needed for context.getResource?

**Comment by guwirth** [comment]
Strange getting this:
```
The command "mvn test" exited with 1.

0.09s$ RAILS_ENV=production PATH=$PATH:/tmp/sonar-runner-2.4/bin SONARHOME=/tmp/sonarqube-4.5.1 behave

Exception SyntaxError: invalid syntax (environment.py, line 65)

Traceback (most recent call last):

File "/usr/local/bin/behave", line 11, in <module>

sys.exit(main())

File "/usr/local/lib/python2.7/dist-packages/behave/__main__.py", line 109, in main

failed = runner.run()

File "/usr/local/lib/python2.7/dist-packages/behave/runner.py", line 672, in run

return self.run_with_paths()

File "/usr/local/lib/python2.7/dist-packages/behave/runner.py", line 677, in run_with_paths

self.load_hooks()

File "/usr/local/lib/python2.7/dist-packages/behave/runner.py", line 631, in load_hooks

exec_file(hooks_path, self.hooks)

File "/usr/local/lib/python2.7/dist-packages/behave/runner.py", line 303, in exec_file

code = compile(f.read(), filename2, 'exec')

File "integration-tests/features/environment.py", line 65

print BRIGHT + "\nSetting up the test environment" + RESET_ALL

^

SyntaxError: invalid syntax
```

**Comment by wenns** [comment]
Most probably youre using python 3. Print is a function there, rather than a keyword as in 2.x series. I'll change the scripts to be python 3 compiant. In the meantime, you could use either use python 2 or change those lines yourself in something like 
```Python
from __future__ import print_function
...
print(BRIGHT + "\nSetting up the test environment" + RESET_ALL)
```

**Comment by wenns** [comment]
Seems like they're using python 3 on Travis now. I'll change the syntax right now.

**Pull request merged**